lib/ovirt/service.rb

```diff
@@ -217,13 +227,19 @@ def api_uri(path = nil)
 
     def self.ovirt?(options)
       options[:username] = options[:password] = "_unused"
-      !new(options).engine_ssh_public_key.to_s.blank?
-    rescue RestClient::ResourceNotFound, NoMethodError
-      false
+      !new(options).engine_ssh_public_key.blank?
     end
 
     def engine_ssh_public_key
-      RestClient::Resource.new("#{base_uri}/engine.ssh.key.txt", resource_options).get
+      CANDIDATE_SSH_PUBLIC_KEY_PATHS.each do |path|
+        begin
+          key = RestClient::Resource.new("#{base_uri}#{path}", resource_options).get
+          return key unless key.blank?
+        rescue RestClient::ResourceNotFound, NoMethodError
+          # Do nothing, just try the next candidate.
+        end
+      end
+      nil
```

Comment:
Sorry for the message spam, but what I mean is, if you don't actually need the key itself, then you can keep the detect above and maybe rename the method to `engine_ssh_public_key?`, then the `ovirt?` method can be

```ruby
     def self.ovirt?(options)
        options[:username] = options[:password] = "_unused"
        new(options).engine_ssh_public_key?
     end
```